liquibase-core/src/main/java/liquibase/integration/servlet/GenericServletListener.java

```diff
@@ -0,0 +1,256 @@
+package liquibase.integration.servlet;
+
+import liquibase.Contexts;
+import liquibase.LabelExpression;
+import liquibase.Liquibase;
+import liquibase.Scope;
+import liquibase.configuration.ConfiguredValue;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.database.DatabaseFactory;
+import liquibase.database.core.DerbyDatabase;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.LiquibaseException;
+import liquibase.integration.commandline.LiquibaseCommandLineConfiguration;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import liquibase.resource.CompositeResourceAccessor;
+import liquibase.resource.FileSystemResourceAccessor;
+import liquibase.resource.ResourceAccessor;
+import liquibase.util.NetUtil;
+import liquibase.util.StringUtil;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Enumeration;
+
+/**
+ * Servlet listener than can be added to web.xml to allow Liquibase to run on every application server startup.
+ * Using this listener allows users to know that they always have the most up to date database, although it will
+ * slow down application server startup slightly.
+ * See the <a href="http://www.liquibase.org/documentation/servlet_listener.html">Liquibase documentation</a> for
+ * more information.
+ *
+ * @see LiquibaseJakartaServletListener
+ * @see LiquibaseServletListener
+ */
+abstract class GenericServletListener {
+
+    private static final String LIQUIBASE_CHANGELOG = "liquibase.changelog";
+    private static final String LIQUIBASE_CONTEXTS = "liquibase.contexts";
+    private static final String LIQUIBASE_LABELS = "liquibase.labels";
+    private static final String LIQUIBASE_DATASOURCE = "liquibase.datasource";
+    private static final String LIQUIBASE_HOST_EXCLUDES = "liquibase.host.excludes";
```

Comment:
@nvoxland, hi - What, if anything, do these parameters correspond to in the CLI? I'm asking in particular about LIQUIBASE_HOST_EXCLUDES and LIQUIBASE_HOST_INCLUDES.

Reply:
They don't correspond to anything in the CLI right now, since when the CLI runs is more controlled by the user. As we continue to clean up the integrations for consistency, the arguments may eventually show up in the CLI but it doesn't so far.

Reply:
Thanks for the explanation.